### PR TITLE
Added Properties message to  RpcFunctionMetadata

### PR DESCRIPTION
**Pull request by gavin-aguiar**
Added  new message to RpcFunctionMetadata called Properties.

This is a property bag where every runtime can add properties specific to them which the host can just parse as a dictionary/ map.

### CODE REVIEW

**Review by satvu — APPROVED** [review]

File: src/proto/FunctionRpc.proto [hunk 1]
```diff
@@ -298,6 +304,12 @@ message RpcFunctionMetadata {
 
   // A flag indicating if managed dependency is enabled or not
   bool managed_dependency_enabled = 14;
+
+  // Indexing mode for the function 
+  FunctionIndexingMode = 15;
```
Comment (surgupta-msft) [inline src/proto/FunctionRpc.proto:309]: Unless I am missing something, we will need to declare variable as - `FunctionIndexingMode function_indexing_mode = 15;`

Comment (surgupta-msft) [reply src/proto/FunctionRpc.proto:309]: Below field also does the same thing if we should think about leveraging it - 
https://github.com/Azure/azure-functions-language-worker-protobuf/blob/6635670b136d565f2a196ce1a64fee1ea5c9c1c9/src/proto/FunctionRpc.proto#L317-L318

File: src/proto/FunctionRpc.proto [hunk 2]
```diff
@@ -298,6 +304,12 @@ message RpcFunctionMetadata {
...
+
+  // Properties for function metadata
+   map<string,string> RpcFunctionMetadataProperties = 16;
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply src/proto/FunctionRpc.proto:312]: In addition, is there a reason the naming here isn't simply `Properties` instead?

**Review by fabiocav — CHANGES_REQUESTED** [review]
Small comments, but there are some open questions from @surgupta-msft as well.

File: src/proto/FunctionRpc.proto (hunk 1 above)
Comment (gavin-aguiar) [reply src/proto/FunctionRpc.proto]: Good point. We can just use the use_default_metadata_indexing property. I have removed the other property which I added.

File: src/proto/FunctionRpc.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (gavin-aguiar) [reply src/proto/FunctionRpc.proto]: @surgupta-msft  This field is for adding properties that are more worker specific which the host does not need to prerequisite info and should be able to parse just this dictionary. 
One of the properties we plan to add in this dictionary is line number of the function which will be used by the portal to separate out different functions from the single function_app file.

@fabiocav  I have renamed this field to _Properties_.

File: src/proto/FunctionRpc.proto [hunk 3]
```diff
@@ -298,6 +298,9 @@ message RpcFunctionMetadata {
 
   // A flag indicating if managed dependency is enabled or not
   bool managed_dependency_enabled = 14;
+
+  // Properties for function metadata
```
Comment (ejizba) [inline src/proto/FunctionRpc.proto:302]: Can we add a bit more context to the comment? For example, if I understand the feature correctly this is perhaps what I would put:

```suggestion
  // Properties for function metadata
  // They're usually specific to a worker and largely passed along to the controller API for use outside the host (e.g. the portal may adjust its experience based on these properties)
```

**Review by ejizba — APPROVED** [review]

**Review by fabiocav — APPROVED** [review]

**Pull request merged**